src/sync/core/events/sensordevice/datastream/added/index.ts

```diff
@@ -1,11 +1,18 @@
 import { plainToClass } from 'class-transformer';
 import { EventMessage } from '../../../event-message';
 import { DatastreamAdded as V1 } from './datastream-added-v1.event';
+import { DatastreamAdded as V2 } from './datastream-added-v2.event';
 
-export { DatastreamAdded } from './datastream-added-v1.event';
+export { DatastreamAdded } from './datastream-added-v2.event';
 
-export function getDatastreamAddedEvent(eventMessage: EventMessage): V1 {
-    return !eventMessage.metadata.version || eventMessage.metadata.version === V1.version
-        ? plainToClass(V1, eventMessage.data)
-        : null;
+export function getDatastreamAddedEvent(eventMessage: EventMessage): V2 {
+    let datastreamAddedEvent = null;
+    if (!eventMessage.metadata.version || eventMessage.metadata.version === V1.version) {
+        const { observationArea, ...data } = eventMessage.data;
```

Comment:
Line 11+12 zijn 'eigenlijk' `upcastV1toV2`, right? Is het wat om deze in een aparte method te stoppen (die zo heet)? Dan is de bedoeling heel duidelijk (... en zouden we die method kunnen voorzien van een unittest 😇 ). Bovendien zou die method ook gebruikt kunnen worden door datastream/updated/index.ts ... want de method is hetzelfde hoewel de body van het json object wel verschilt... toch?

Reply:
Goed idee, ik heb hem gemaakt voor datastreamAdded en voor datastreamUpdated, omdat die events andere types hebben, en als we later 1 van de 2 willen updaten we ze alsnog uit elkaar moeten halen. Eens?